pagopa/io-app
# fix: [IOBP-336] Fix file upload modal in barcodes scan screen

## Short description
This PR fixes the file upload modal shown in the barcodes scan screen. 
The fix includes a refactor of the modal component exposed by the `useIOBarcodeFileReader` hook, moving it outside of the base component and leaving the implementation up to the screen that wants to display it.

## List of changes proposed in this pull request
- Renamed `useIOBarcodeFileScanner` to `useIOBarcodeFileReader`
- Removed `useIOBarcodeFileReader` from inside the `BarcodeScanBaseScreenComponent` and moved to every screen which needs to display it

## How to test
On the barcodes scan screen, tap on "Upload" button and check that the modal buttons are responsive to taps.
